### PR TITLE
hookup siginin to backend

### PR DESCRIPTION
**Pull request by EngrEric**

### CODE REVIEW

File: src/components/signin/sagas.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { effects } from 'redux-saga';
+import {  SIGNIN_REQUEST } from './actionTypes';
+import { SIGNIN_URL } from './constants';
+import { signinSuccess, signinFailure } from './actions';
+
```
Comment (enyeInc) [inline src/components/signin/sagas.js]: add js comment to function

File: src/components/signin/sagas.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        if(response.ok){
+            const data = yield response.json();
+            yield effects.put(signinSuccess(data));
+        }
+    } catch(error){
+        yield effects.put(signinFailure(error));
+    }
+}
+
+function* watchSigninUser(){
```
Comment (enyeInc) [inline src/components/signin/sagas.js]: add comment to function

File: src/components/signin/actionTypes.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export const SIGNIN_REQUEST = "SIGN_REQUEST";
+export const SIGNIN_SUCCESS = "SIGNIN_SUCCESS";
+export const SIGNIN_FAILURE = "SIGNIN_FAILURE";
```
Comment (Way2nnadi) [inline src/components/signin/actionTypes.js]: comment your `actionTypes`

File: src/components/signin/selectors.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { NAME } from './constants';
+
+export const getStatus = state => state[NAME].isSigninSuccessful;
+
+export const getMessage = state => state[NAME].message;
```
Comment (Way2nnadi) [inline src/components/signin/selectors.js]: add comments to your selectors. Example:
```
/**
 * Selects the <tt>isExample</tt> key.
 *
 * @function
 * @param {Object} state - redux store state
 * @return {Boolean} the state of the example
 * {@link module:example/constants::INITIAL_STATE constants::INITIAL_STATE}).
 */
export const getExampleState = state => state[NAME].isExample;
```

File: src/components/signup/selectors.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { NAME } from './constants';
+
+export const getStatus = state =>  state[NAME].isSignupSuccessful;
+
+export const getMessage = state => state[NAME].message;
```
Comment (Way2nnadi) [inline src/components/signup/selectors.js]: add comment to your selectors (see example above).

File: src/components/signup/components/Signup.js [hunk 6]
```diff
@@ -61,6 +65,16 @@ class Signup extends React.Component {
...
+
+                {
+                    signupStatus !== undefined &&
+                    <span>
+                        {
+                            signupStatus ? message.success(signupMessage, 5) :
+                                message.error(signupMessage, 5)
+                        }
+                    </span>
+                }
```
Comment (Simpcyclassy) [inline src/components/signup/components/Signup.js]: Oh! I was looking for where this logic went to from home.js

**Comment by joshuaobinna** [comment]
well done @EngrEric

File: src/components/signup/components/Signup.js (hunk 6 above)
Comment (EngrEric) [reply src/components/signup/components/Signup.js]: I moved them to there individual components instead of having it at home

**Comment by Way2nnadi** [comment]
@EngrEric Looks good to me

**Pull request merged**